src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -144,11 +167,27 @@ def validate_interval(self, interval):
         return interval
 
     def validate(self, data):
+        is_issue_widget = data.get("widget_type") == DashboardWidgetTypes.ISSUE
+        query_errors = []
+        has_query_error = False
         if not data.get("id"):
             if not data.get("queries"):
                 raise serializers.ValidationError(
                     {"queries": "One or more queries are required to create a widget"}
                 )
+            # Check each query to see if they have an issue or discover error depending on the type of the widget
+            for query in data.get("queries"):
+                if is_issue_widget and "issue_query_error" in query:
+                    query_errors.append(query["issue_query_error"])
+                    has_query_error = True
+                elif not is_issue_widget and "discover_query_error" in query:
+                    query_errors.append(query["discover_query_error"])
+                    has_query_error = True
+                else:
+                    query_errors.append({})
+            if has_query_error:
+                raise serializers.ValidationError({"queries": query_errors})
```

Comment:
Here we evaluate the query errors that are bubbled up to `DashboardWidgetSerializer` from `DashboardWidgetQuerySerializer`. If our widget is an `issue` widget, we only consider `issue` query errors. If our widget is a `discover` widget, we only consider `discover` query errors.